xbmc/addons/interfaces/GUI/controls/Slider.cpp

```diff
@@ -0,0 +1,266 @@
+                          __FUNCTION__, addon, control, addon ? addon->ID().c_str() : "unknown");
+    return;
+  }
+
+  control->SetEnabled(enabled);
+}
+
+void Interface_GUIControlSlider::reset(void* kodiBase, void* handle)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUISliderControl* control = static_cast<CGUISliderControl*>(handle);
+  if (!addon || !control)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIControlSlider::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, control, addon ? addon->ID().c_str() : "unknown");
+    return;
+  }
+
+  CGUIMessage msg(GUI_MSG_LABEL_RESET, control->GetParentID(), control->GetID());
+  g_windowManager.SendThreadMessage(msg, control->GetParentID());
+}
+
+char* Interface_GUIControlSlider::get_description(void* kodiBase, void* handle)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUISliderControl* control = static_cast<CGUISliderControl*>(handle);
+  if (!addon || !control)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIControlSlider::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, control, addon ? addon->ID().c_str() : "unknown");
+    return nullptr;
+  }
+
+  return strdup(control->GetDescription().c_str());
+}
+
+void Interface_GUIControlSlider::set_int_range(void* kodiBase, void* handle, int start, int end)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUISliderControl* control = static_cast<CGUISliderControl*>(handle);
+  if (!addon || !control)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIControlSlider::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, control, addon ? addon->ID().c_str() : "unknown");
+    return;
+  }
+
+  control->SetType(SLIDER_CONTROL_TYPE_INT);
+  control->SetRange(start, end);
+}
+
+void Interface_GUIControlSlider::set_int_value(void* kodiBase, void* handle, int value)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUISliderControl* control = static_cast<CGUISliderControl*>(handle);
+  if (!addon || !control)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIControlSlider::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, control, addon ? addon->ID().c_str() : "unknown");
+    return;
+  }
+
+  control->SetType(SLIDER_CONTROL_TYPE_INT);
+  control->SetIntValue(value);
+}
+
+int Interface_GUIControlSlider::get_int_value(void* kodiBase, void* handle)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUISliderControl* control = static_cast<CGUISliderControl*>(handle);
+  if (!addon || !control)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIControlSlider::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, control, addon ? addon->ID().c_str() : "unknown");
+    return 0;
```

Comment:
"return -1;" to indicate error, like done in all (?) other methods returning int?